DRODLib/Character.cpp

```diff
@@ -5093,6 +5214,57 @@ void CCharacter::SetVariable(const CCharacterCommand& command, CCurrentGame* pGa
 	}
 }
 
+//*****************************************************************************
+void CCharacter::SetArrayVariable(
+	const CCharacterCommand& command,
+	CCurrentGame* pGame,
+	CCueEvents& CueEvents)
+{
+	const UINT varIndex = command.w;
+	if (pGame->pHold && !pGame->pHold->IsArrayVar(varIndex))
+		return; //Don't set normal var as an array var
+
+	int arrayIndex = (int)(this->paramF == NO_OVERRIDE ? command.flags : this->paramF);
+	ScriptArrayMap& scriptArrays = pGame->pHold->IsLocalVar(varIndex) ? this->localScriptArrays : pGame->scriptArrays;
+
+	vector<WSTRING> expressions = WCSExplode(command.label, *wszSemicolon);
+	for (vector<WSTRING>::const_iterator expression = expressions.begin();
+		expression != expressions.end(); ++expression) {
+		UINT index = 0;
+		int x = scriptArrays[varIndex][arrayIndex];
```

Comment:
We should probably note with a comment that operator[] here will implicitly add a value to the map if the key doesn't already exist.